projects/plugins/jetpack/modules/carousel/jetpack-carousel.js

```diff
@@ -43,8 +43,45 @@
 				return;
 			}
 
+			// Adjust the canvas size.
+			canvas.width = window.innerWidth;
+			canvas.height = window.innerHeight;
+
+			var imageRatio = imgEl.naturalHeight / imgEl.naturalWidth;
+			var winRatio = window.innerHeight / window.innerWidth;
+
+			// Calculate adjusted image dimensions to cover the canvas.
+			var imageHeight = imgEl.naturalHeight;
+			var imageWidth = imgEl.naturalWidth;
+			if ( imageRatio > winRatio ) {
+				imageHeight = imgEl.naturalWidth * winRatio;
+			} else {
+				imageWidth = imgEl.naturalHeight / winRatio;
+			}
+
+			// Apply image blur.
+			if ( context.filter !== undefined ) {
+				context.filter = 'blur(20px) ';
+			} else {
+				// Class used to apply CSS blur on Safari, where it is not supported by the canvas.
+				// Canvas blur is preferred for performance reasons on Chrome.
+				document
+					.querySelector( '.jp-carousel' )
+					.classList.add( 'jp-carousel-canvas-blur-unsupported' );
+			}
+
 			context.filter = 'blur(20px) ';
-			context.drawImage( imgEl, 0, 0 );
+			context.drawImage(
+				imgEl,
+				( imgEl.naturalWidth - imageWidth ) * 0.5,
+				( imgEl.naturalHeight - imageHeight ) * 0.5,
+				imageWidth,
+				imageHeight,
+				0,
+				0,
+				window.innerWidth,
+				window.innerHeight
+			);
```

Comment:
Interesting -- I can't reproduce this 🤔 It's tough because I think if you're going to do blurred backgrounds, it makes the most sense to get as close as possible to `background-size: cover` -- at any rate that's what other implementations of this feature seem to do. I'm trying here to also center the zoomed-in portion. This definitely ends up creating fidgety logic. If we move forward with this I'll take another look.